Tests/OpenAPIRuntimeTests/URICoder/Serialization/Test_URISerializer.swift

```diff
@@ -103,7 +109,8 @@ final class Test_URISerializer: Test_Runtime {
                     simpleExplode: "red,green,blue",
                     simpleUnexplode: "red,green,blue",
                     formDataExplode: "list=red&list=green&list=blue",
-                    formDataUnexplode: "list=red,green,blue"
+                    formDataUnexplode: "list=red,green,blue",
+                    deepObjectExplode: "list=red&list=green&list=blue"
```

Comment:
Let's figure out what we do with arrays, I think either we don't support them, or they should be something like `list[]=red&...` or `list[0]=red&...`

Reply:
Not sure, I'd like to see what other generators are doing. Also, @simonjbeaumont, any thoughts?

Reply:
I created a very simple example on [swagger](https://editor-next.swagger.io).

**Yaml file:**
```
---
openapi: 3.0.0
info:
  title: DeepObject Example
  version: TODO
servers:
- url: http://localhost:3000
  description: Development
paths:
  "/example":
    get:
      responses:
        '200':
          description: OK
      parameters:
        - in: query
          name: include
          schema:
            type: object
            properties:
              member:
                type: array
                example: [foo, bar]
          style: deepObject
          explode: true
  
```

**Generated request:**
```
http://localhost:3000/example?include[member]=foo&include[member]=bar
```

I don't know if that helps, it didn't helped me ..

Reply:
Unless we can find some stronger evidence for how it should be, I'd rather us not support it at all.

I took a quick look at `opai-codegen` and it seems they only test the object case (cf. array)[^1].

[^1]: https://github.com/deepmap/oapi-codegen/blob/dd082985a9b6e8f68472987f6c2c60fea5e59871/internal/test/parameters/parameters_test.go#L430-L436

Reply:
> ```
> http://localhost:3000/example?include[member]=foo&include[member]=bar
> ```

Oh, from the example above, it looks like arrays are serialized as repeated values for the same key. That's gonna complicate the logic some, I think. Be good to see if we can find any documentation that supports this.

Reply:
Ok let's just not support non-object values when deepObject is used, and throw appropriate errors. We can always add it later once we get more clarity.